src/components/SidePanel/StartStop.tsx

```diff
@@ -88,8 +91,28 @@ export default () => {
             mode,
             samplesPerSecond: DataManager().getSamplesPerSecond(),
         });
-        dispatch(samplingStart());
+
+        const space = Math.max(
+            0,
+            await getFreeSpace(diskFullTrigger, sessionFolder)
+        );
+
+        setFreeSpace(space);
+
+        if (space === 0) {
+            logger.warn('Disk is full. Unable to start new session');
```

Comment:
```suggestion
            logger.warn('Disk is full. Unable to start sampling. Try changing disk full trigger threshold or free disk memory.');
```
@kylebonnici @greg-fer what do you think?

Reply:
@greg-fer  I have changed it to @KievDevel  recommendation let us know if you agree :)

Reply:
``Disk is full. Unable to start sampling. Change the disk full trigger threshold or free up disk memory.``